### DISCUSSION TITLE
Update terrain handling to the 1.17/1.18

### DISCUSSION THREAD

**Discussion by Daenges** [ORelio/Minecraft-Console-Client#1895] [Q&A] [answered]
I want to ask, wheather it is possible to update the terrain handling to the most recent versions.
As I saw in two of the three pull requests, either a lack of documentation or the .NET 5.0 support currently block their implementation.

@ReinforceZwei as you worked on the terrain handling process for 1.16: Do you know whether something currently speaks against an implementation? And what would need to be done to implement it?

**Comment by ReinforceZwei** [comment, marked as answer]
The main problem at present is the lack of contributors. And compared to before, I don't have enough time to work on the project due to private work. Advance features like inventory, entity, terrain and movement got left behind to ensure basic function can be implemented as soon as possible. It's hard to tell when we could update those advance features for now.

**Reply by Daenges** [reply]
Maybe I could look into movement. Your PR suggests that it is mainly addition to materials and paletts, with further improvements, that seem to focus on the overall perfromance.

Could you maybe provide some hints, so I can try looking into it?

**Reply by ReinforceZwei** [reply]
You have to first find out protocol changes between 1.16 and 1.17, implement the changes. Then, update material enum and generate block palette. There's a [script](https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/Mapping/BlockPalettes/BlockPaletteGenerator.cs) for generating them. Remember to test it and make sure everything works fine.

You may take #1353 as an example.

**Reply by ORelio** [reply]
Same for me, I've not been playing Minecraft for years and have many other commitments than working on a project I no longer use.
Any contributor is welcome, we can provide guidance, answer questions and review pull requests. 🙂
